### DISCUSSION TITLE
Keine doppelte Anzeige von Diensten bei "Inhalte suchen/hinzufügen"

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/webgis-community#124] [General]
Hallo!

Wir führen manche Karten doppelt (z.B. Gefahrenzonen im Reiter "Planung und Kataster" und im Reiter "Wasser", DKM im Reiter "Basiskarten" und im Reiter "Planung & Kataster"). 
![image](https://github.com/gis-eni/webgis-forum/assets/127403099/f97076b4-0f08-498e-bbf0-d42118d1aec5)

Diese Reiterstruktur haben wir bei den hinzufügbaren Diensten ebenfalls:
![image](https://github.com/gis-eni/webgis-forum/assets/127403099/d1867957-5632-4f2c-9d2e-24f9b95c47e2)

Wir haben im CMS die Gefahrenzonen also zweimal als hinzufügbaren Dienst eingetragen. Einmal im Reiter Planung und Kataster, einmal im Reiter Wasser. In der Anwendung selbst scheinen die Gefahrenzonen im Reiter Wasser nicht mehr auf:
![image](https://github.com/gis-eni/webgis-forum/assets/127403099/4592a638-960c-4d35-ac70-95ce5c0c599e)

Das darf unserer Meinung nicht sein. Entweder ich kann wirklich nur **einen** Dienst als hinzufügbaren Dienst eintragen (dann muss beim Versuch den Dienst im CMS nochmals hinzuzufügen aber ein Hinweis kommen!), oder es ist möglich einen Dienst mehrmals als hinzufügbaren Dienst einzutragen. Zweiteres macht unserer Meinung nach mehr Sinn, da die User oft von einer bestimmten Fachrichtung kommen und dann verwirrt sind warum der Dienst nicht im ihnen fach-nahen Reiter aufscheint (bzw. sie nicht in einem anderen Reiter danach suchen müssen).

**Comment by gis-eni** [comment]
Das CMS kann solche "Quer" Abgleiche leider nicht abfangen. Da alles im Hintergrund in einem Filesystem Baum abgelegt wird, kann nicht bei jedem Einfügen (=verknüpfen, verlinken) eines Knotens der ganze Baum auf Fehler durchsucht werden. Das wäre nicht sehr performant.

Beim Publizieren gibt es zwar "WARNINGS", allerdings werden hier nur Verknüpfungen überprüft, die ins Leere zeigen. Wenn du beispielsweise einen Dienst unter hinzufügbare Dienste einträgst und den Dienst dann löscht, bleibt die Verlinkung erhalten (aus dem selben Grund wie oben, es kann nicht immer der komplette Baum durchsucht werden). Da bekommst du beim Veröffentlichen des CMS allerdings die Warnung...

Es ist hier recht schwierig, alles zu überprüfen, was dann der Viewer zulässt und was nicht. In diesem Fall ist es leider so, dass ein Dienst nur einmal in einer Karte vorkommen darf. Das gilt dann ebenfalls für die hinzufügbaren Dienste!

Das zu Ändern wäre leider ein *Breaking Change* (sprich das könnte sich auf andere Installation auswirken). Darum bleibt die Regel fürs erste einmal aufrecht.

Zur Usability:
Wir verwenden hinzufügbare Dienste auch recht intensiv. Und haben auch viele Untergruppen, hatten aber noch nie den Wunsch, einen Dienst öfter einzubinden. Ganz im Gegenteil!
Der Anwender sollte eigentlich wissen, aus welcher Sparte er einen Dienst bezieht. Wenn wir den Gas Leitungsdienst auch in der Kategorie *Strom" anbieten würden, wäre die Verwirrung noch größer... 

* Was ist der Unterschied zwischen den beiden Diensten!? 
* Soll ich den "richtigen" Gas Dienst nehmen oder den für "Strom"?  

Der Anwender wählt bei uns bewusst den Dienst aus der Sparte aus, auf die er zugreifen möchte (und berechtigt ist). Es gibt im Dialog ja auch die Suche dafür.
